### DISCUSSION TITLE
5-ways-to-set-the-urls-for-an-aspnetcore-app/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#81] [Announcements]
# 5-ways-to-set-the-urls-for-an-aspnetcore-app/

In this post I describe 5 different ways to set which URLs your ASP.NET Core application listens on.

https://andrewlock.net/5-ways-to-set-the-urls-for-an-aspnetcore-app/

**Comment by zoran-petrovic-87** [comment]
Great and detailed article! However, in .NET 6 for some reason none of these works for me, except UseUrls(). In all other cases I get "Now listening on: http://localhost:5000". Did they changed something in .NET6?

**Comment by SalimGangji** [comment]
There is now another option for configuring Kestrel, through `appsettings.json`

For example if you want to listen on port 5110

```json
{
  "Kestrel": {
    "EndPoints": {
      "Http": {
        "Url": "http://0.0.0.0:5110"
      }
    }
  }
}
```

**Reply by alexdresko** [reply]
For completeness sake, I really wish the article was updated with this information.

**Reply by mpawelski** [reply]
Yes, I wasted too much time figuring out that this is the one used by my project (.NET 6). I only found it by using "GetDebugView" to get all configuration values and searching for the port used 🤷

@andrewlock Please update the article if you can ;)

**Reply by andrewlock** [reply]
This post is 4.5 years old, there's an updated version here: https://andrewlock.net/8-ways-to-set-the-urls-for-an-aspnetcore-app/ 😊

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at April 26 2020, 10:01, in reply to [Rahul Sharma](https://disqus.com/by/disqus_0JjNBWCu48/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/81#discussioncomment-6058468)</em>

---

<p>Hi Rahul, thanks:) No it hasn't been updated to support version 3.1. The majority of the book still applies though, the biggest changes being around endpoint routing.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at April 26 2020, 10:07, in reply to [Giorgio Brantzos](https://disqus.com/by/giorgiobrantzos/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/81#discussioncomment-6058473)</em>

---

<p>A good point, I'll have to update it to "6 ways" :D</p>

[a comment by an automated account is not shown]

**Comment by mahmah2** [comment]
Good article!
Thanks
